library/psa_crypto.c

```diff
@@ -315,7 +349,9 @@ psa_status_t mbedtls_to_psa_error(int ret)
             return PSA_ERROR_BUFFER_TOO_SMALL;
         case MBEDTLS_ERR_RSA_RNG_FAILED:
             return PSA_ERROR_INSUFFICIENT_ENTROPY;
+#endif
 
+#if defined(MBEDTLS_ECP_C)
```

Comment:
I wonder if we should use `MBEDTLS_ECP_LIGHT` here? `MBEDTLS_ECP_LIGHT` would be auto-enabled by `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR_DERIVE` and some of the functions might return `MBEDTLS_ERR_ECP_xxx` error codes.

Reply:
Yes, that's a good spot - all three of these are returned by functions in `ecp.c` that are under `MBEDTLS_ECP_LIGHT` but not `MBEDTLS_ECP_C` @daverodgman this needs to be updated